### DISCUSSION TITLE
timeit equivalent

### DISCUSSION THREAD

**Discussion by superlopuh** [marimo-team/marimo#552] [Q&A] [answered]
What's the best way to time the time to execute a function, like `% timeit` in Jupyter?
This didn't work for me, I assume beacuse Marimo messes with what `__main__` is:

``` python
import timeit
def one():
    return 1
timeit.repeat("one()", "from __main__ import one", number=100000)
```

**Comment by superlopuh** [comment, marked as answer]
Oh this seems to work:

``` python
import timeit

def bla():
    return 1
timeit.repeat("bla()", globals=globals())
```

**Reply by akshayka** [reply]
Oh interesting -- I didn't realize that `timeit` didn't work out of the box. Thanks for letting us know. We'll add this to our docs/FAQs. Thinking aloud, we could also add a convenience function `mo.timeit.repeat()` that just partials `timeit.repeat` and automatically supplies the globals, and similarly for the other functions in timeit.
